plugin/src/android/mainApplicationDependency.ts

```diff
@@ -0,0 +1,43 @@
+import { ConfigPlugin, withMainApplication } from '@expo/config-plugins'
+
+import { PluginConfigType } from '../pluginConfig'
+
+/**
+ * Update `<project>/build.gradle` by adding the codepush.gradle file
+ * as an additional build task definition underneath react.gradle
+ */
+
+function applyImplementation(
+  mainApplication: string,
+  find: string,
+  add: string,
+  replace?: boolean
+) {
+  // Make sure the project does not have the settings already
+  if (!mainApplication.includes(add)) {
+    if (replace) return mainApplication.replace(find, add)
+    else return mainApplication.replace(find, `${find}\n${add}`)
+  }
+
+  return mainApplication
+}
+
+export const withAndroidMainApplicationDependency: ConfigPlugin<PluginConfigType> = (config) => {
+  return withMainApplication(config, (config) => {
+    config.modResults.contents = applyImplementation(
+      config.modResults.contents,
+      'import expo.modules.ReactNativeHostWrapper;',
+      `\nimport com.microsoft.codepush.react.CodePush;`
+    )
+    config.modResults.contents = applyImplementation(
+      config.modResults.contents,
+      `new ReactNativeHost(this) {`,
+      `
+    @Override
+    protected String getJSBundleFile() {
+      return CodePush.getJSBundleFile();
+    }\n`
+    )
+    return config
```

Comment:
Apply the case for react native 71 (see : https://react-native-community.github.io/upgrade-helper/?from=0.70.9&to=0.71.8#RnDiffApp-android-app-src-main-java-com-rndiffapp-MainActivity.java)
```suggestion
    if (
      config.modResults.contents.includes("new DefaultReactNativeHost(this) {")
    ) {
      config.modResults.contents = applyImplementation(
        config.modResults.contents,
        `new DefaultReactNativeHost(this) {`,
        `
          @Override
          protected String getJSBundleFile() {
            return CodePush.getJSBundleFile();
          }\n`
      );
    } else if (
      config.modResults.contents.includes(" new ReactNativeHost(this) {")
    ) {
      config.modResults.contents = applyImplementation(
        config.modResults.contents,
        `new ReactNativeHost(this) {`,
        `
    @Override
    protected String getJSBundleFile() {
      return CodePush.getJSBundleFile();
    }\n`
      );
    }
    return config;
```